File: edgequake/sdks/typescript/src/resources/chunks.ts
/**
 * Chunks resource — chunk-level detail access.
 *
 * WHY: Updated to use proper ChunkDetailResponse from lineage.ts.
 * Matches Rust ChunkDetailResponse with rich metadata (char_range,
 * entities, relationships, extraction_metadata).
 *
 * @module resources/chunks
 * @see edgequake/crates/edgequake-api/src/handlers/chunks.rs
 */

import type { ChunkDetailResponse, ChunkLineageResponse } from "../types/lineage.js";
import { Resource } from "./base.js";

export class ChunksResource extends Resource {
  /** Get chunk details by ID. */
  async get(chunkId: string): Promise<ChunkDetailResponse> {
    return this._get(`/api/v1/chunks/${chunkId}`);
  }

  /**
   * Get chunk lineage with parent document refs and position info.
   * @implements F3 — Every chunk contains parent_document_id and position info.
   * @implements F8 — PDF → Document → Chunk → Entity chain traceable.
   */
  async getLineage(chunkId: string): Promise<ChunkLineageResponse> {
    return this._get(`/api/v1/chunks/${chunkId}/lineage`);
  }
}

File: edgequake/sdks/rust/src/resources/chunks.rs
//! Chunks resource.

use crate::client::EdgeQuakeClient;
use crate::error::Result;
use crate::types::operations::{ChunkDetail, ChunkLineageInfo};

pub struct ChunksResource<'a> {
    pub(crate) client: &'a EdgeQuakeClient,
}

impl<'a> ChunksResource<'a> {
    /// `GET /api/v1/documents/{doc_id}/chunks`
    pub async fn list(&self, document_id: &str) -> Result<Vec<ChunkDetail>> {
        self.client
            .get(&format!("/api/v1/documents/{document_id}/chunks"))
            .await
    }

    /// `GET /api/v1/chunks/{id}`
    pub async fn get(&self, id: &str) -> Result<ChunkDetail> {
        self.client.get(&format!("/api/v1/chunks/{id}")).await
    }

    /// `GET /api/v1/chunks/{id}/lineage`
    ///
    /// Returns chunk lineage with parent document refs and position info.
    /// @implements F3 — Every chunk contains parent_document_id and position info.
    /// @implements F8 — PDF → Document → Chunk → Entity chain traceable.
    pub async fn get_lineage(&self, id: &str) -> Result<ChunkLineageInfo> {
        self.client
            .get(&format!("/api/v1/chunks/{id}/lineage"))
            .await
    }
}
